components/raftstore/src/store/fsm/store.rs

```diff
@@ -242,10 +242,17 @@ where
         }
         let peer_msg = PeerMsg::RaftMessage(InspectedRaftMessage { heap_size, msg });
         let event = TraceEvent::Add(heap_size);
+        let send_failed = Cell::new(true);
+
+        MEMTRACE_RAFT_MESSAGES.trace(event);
+        defer!(if send_failed.get() {
+            MEMTRACE_RAFT_MESSAGES.trace(TraceEvent::Sub(heap_size));
+        });
```

Comment:
What is the difference with the original code?

Reply:
Original code logic is:
    op1: send msg; op2: add size.
If the msg handle occurs between op1 and op2, then the sub size will happen before add size, which leads to overflow.

Reply:
Got it